core/src/main/java/io/cloudevents/core/CloudEventUtils.java

```diff
@@ -102,4 +102,16 @@ public static CloudEvent toEvent(CloudEventReader reader, @Nullable CloudEventDa
         return mapper.map(data);
     }
 
+    /**
+     * Validates the extension name as defined in  CloudEvents spec
+     * @param name the extension name
+     * @return true if extension name is valid, false otherwise
+     */
+    public static boolean isValidExtensionName(String name) {
+        if (name.matches("[a-z0-9]+") && name.length() <= 20) {
```

Comment:
I wonder, can we avoid using a regexp here? Because the assertion is pretty simple, maybe we can avoid the regexp overhead by doing a char by char comparison, like for each `char c`: `(c >= 'a' && c <= 'z') || (c >= '0' && c <= '9')`